crates/services/txpool_v2/src/service.rs

```diff
@@ -764,6 +796,7 @@ where
     let pruner = TransactionPruner {
         txs_ttl: config.max_txs_ttl,
         time_txs_submitted: Arc::new(RwLock::new(VecDeque::new())),
+        height_expiration_txs: Arc::new(RwLock::new(BTreeMap::new())),
```

Comment:
can't we use dashmap here?
will prevent usage of `RwLock`

also may be worth to think about usage of channels to send transactions that need to be pruned and then the `TransactionPruner` can handle them appropriately according to some configuration

Reply:
Yeah we can. We didn't use it across the code yet but could be a great first place experiment but is it really useful to it if we want to change to channels etc (which could a great idea that I need to study). Are you ok if I just open an issue about improving the transaction pruner ?

Reply:
sure, feel free to do that :)